### PR TITLE
Unblock documentation build

### PR DESCRIPTION
**Pull request by milesgranger**
- [ ] Part of #10805 
- [x] Tests added / passed
- [x] Passes `pre-commit run --all-files`

Just restricting the newish slew of releases from sphinxcontrib which made the `sphinx>=5` start failing. Feel free to merge this now, while I still poke at who's causing the <=4.5.0 restriction.

---

Caused by `dask-sphinx-theme` pinned to `sphinx-book-theme==0.2.0` which requires sphinx<5.
xref https://github.com/dask/dask-sphinx-theme/issues/68

### CODE REVIEW

**Comment by milesgranger** [comment]
@phofl I'm not sure if we want https://github.com/dask/dask/pull/10807/commits/7c14d41a3d9bdcc79b7de73b4f4b46a752dc0821, but as it's not a library, pinning everything seems like an easy way to protect against future breakages from others upgrading their dependencies.

**Comment by milesgranger** [comment]
Tried with mamba: https://readthedocs.org/projects/dask/builds/23132383/
but not happy. Can try again in a follow-up, but using pip with the pinned packages works for now at least.

<details>
<summary>environment.yml</summary>

```
name: dask-docs
channels:
  - conda-forge
dependencies:
  - python=3.11
  - cloudpickle>=1.5.0
  - pandas>=1.4.0
  - sphinx>=4.0.0
  - numpydoc
  - dask-sphinx-theme>=3.0.0
  - sphinx-click
  - sphinx-copybutton
  - sphinx-remove-toctrees
  - sphinx-design
  - jupyter_sphinx
  - toolz
  - fsspec
  - scipy
  - pytest
  - pytest-check-links
  - requests-cache
  - ipython
  - ipykernel<6.22.0
  - pip
  - pip:
      - sphinx-tabs
      - sphinx_autosummary_accessors
      - git+https://github.com/dask/distributed
```

</details>

**Comment by crusaderky** [comment]
-1 from me about pip-compile. Past experience with exact pins has been unexpectedly painful for me.
Can we just add the 5 new restrictions on sphinxcontrib to requirements-doc.txt?

**Comment by milesgranger** [comment]
> -1 from me about pip-compile. Past experience with exact pins has been unexpectedly painful for me.
Can we just add the 5 new restrictions on sphinxcontrib to requirements-doc.txt?

Sure thing, no problem. Out of curiosity, do you remember what your pains have been? W/o pinning application requirements I find myself doing this little number every now and again. My experience with fully pinned requirements outside of libs have been rather pleasant.

File: docs/requirements-docs.txt [hunk 1]
```diff
@@ -8,6 +8,11 @@ sphinx_autosummary_accessors
 sphinx-tabs
 sphinx-design
 jupyter_sphinx
+sphinxcontrib-applehelp>=1.0.0,<1.0.7
```
Comment (jrbourbeau) [inline docs/requirements-docs.txt:11]: A comment here with a little more context would be good (mostly so we know when we can remove these pins)

```suggestion
# FIXME: `sphinxcontrib-*` pins are a workaround until we have sphinx>=5.
#        See https://github.com/dask/dask-sphinx-theme/issues/68.
sphinxcontrib-applehelp>=1.0.0,<1.0.7
```

**Review by jrbourbeau — COMMENTED** [review]
Thanks @milesgranger! One minor comment about a comment

File: docs/requirements-docs.txt (hunk 1 above)
Comment (jrbourbeau) [reply docs/requirements-docs.txt:13]: I went ahead and committed this minor change (hope that's okay @milesgranger)

**Review by jrbourbeau — APPROVED** [review]

**Comment by crusaderky** [comment]
> Sure thing, no problem. Out of curiosity, do you remember what your pains have been? W/o pinning application requirements I find myself doing this little number every now and again. My experience with fully pinned requirements outside of libs have been rather pleasant.

My experience, specifically with conda, is that builds can be deleted by the author or by maintenance bots and it's happened to me many times that an exact-pin conda environment (output of `conda env export`) will just stop working after a while because of this.
I don't have first-hand experience with pip.

**Pull request merged**